proposals/0012-introduce-reactNativeMetadata.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Introducing `reactNativeMetadata` to `package.json`, for RN specific metadata
+author:
+  - Lorenzo Sciandra
+date: 25-01-2021
+---
+
+# RFC0012: Introducing `reactNativeMetadata` to `package.json`, for RN specific metadata
+
+## Summary
+
+This RFC wants to introduce and formally discuss a new section for the package.json of a react-native project (app or library), called `reactNativeMetadata`.
+
+This new section will allow developer to express certain characteristics of their code in a more formalized manner, that can easily used by tooling to act towards the code accordingly.
+
+**Note well**: this is but the first draft of the actual shape and fields. The goal of this draft is to collect feedback and help finalize the spec for this section.
+
+## Basic examples
+
+This is how this section would look like...
+
+...for a library:
+
+```json
+{
+    "name": "@rnx-kit/metro-serializer",
+    "version": "1.0.11",
+    ...,
+    "reactNativeMetadata": {
+        "version": "2.3",
+        "type": "library",
+        "features": {
+            "fabric": "true",
+            "android": {
+                "turbomodules": "true",
+            },
+        },
+        "requirements": {
+            "react-native": ">= 0.71.0",
+            "expo": ">= 46.0.0",
+            "expo-modules-core": "1.2.0"
+        }
+    }
+}
+```
+
+...for an app:
+
+```json
+{
+    "name": "contoso",
+    "version": "2.0.3",
+    ...,
+    "reactNativeMetadata": {
+    "version": "1.0",
+    "type": "app",
+    "features": {
```

Comment:
I'm not sure that `features` here makes as much sense as it does in the library scenario.

Reply:
Resolving as this is now `capabilities`